lte/gateway/python/magma/subscriberdb/store/sqlite.py

```diff
@@ -84,6 +110,50 @@ def edit_subscriber(self, subscriber_id):
                               "WHERE subscriber_id = ?",
                               (data_str, subscriber_id))
 
+    @contextmanager
+    def edit_subscriber_apn(self, sid, request):
+        """
+        Context manager to modify the apn data for a subscriber.
+        """
+        with self.conn:
+            res = self.conn.execute(
+                "SELECT data FROM subscriberdb WHERE " "subscriber_id = ?",
+                (sid,),
+            )
+            row = res.fetchone()
+            if not row:
+                raise SubscriberNotFoundError(sid)
+            non_3gpp = Non3GPPUserProfile()
+            sub_data = self.get_subscriber_data(sid)
+            num_apn = len(request.non_3gpp.apn_config)
+            for idx in range(num_apn):
+                res = self.conn.execute(
+                    "SELECT data FROM apndb WHERE " "apn_name = ?",
+                    (request.non_3gpp.apn_config[idx].service_selection,),
+                )
+
+                row = res.fetchone()
+                if not row:
+                    raise ApnNotFoundError()
+
+                apn_config = non_3gpp.apn_config.add()
+                apn_config.service_selection = request.non_3gpp.apn_config[
+                    idx
+                ].service_selection
```

Comment:
You must fill in all the fields for apn configuration.

Reply:
This is the subscriberdb table which contains only the APN name. All other APN details are present in the apnDB